File: cosmol_viewer/examples/render_molecule_stick.py
from cosmol_viewer import Molecule, Scene, Viewer

mol_data = open("./examples/6fi1_ligand.sdf", "r", encoding="utf-8").read()

mol = Molecule.from_sdf(mol_data).centered().stick().enable_outline(width=0.04)

scene = Scene()
scene.set_scale(1.2)

scene.add_shape_with_id("molecule", mol)
scene.set_camera_view(azimuth=-55, elevation=20, distance=32, fov=18)

scene.save_image("render_molecule_stick.png", width=1200, height=900)

viewer = Viewer.render(scene, width=800, height=500)

print("Press Any Key to exit...", end="", flush=True)
try:
    _ = input()
except EOFError:
    pass

File: cosmol_viewer/examples/render_molecule_stick.rs
use cosmol_viewer::{Scene, Viewer, shapes::Molecule};
use std::path::Path;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mol = Molecule::from_sdf(include_str!("../examples/6fi1_ligand.sdf"))?
        .centered()
        .stick()
        .enable_outline(0.04);

    let mut scene = Scene::new();

    scene.add_shape_with_id("mol", mol);
    scene.set_camera_view(35.0, 20.0, 0.0, 32.0, [0.0, 0.0, 0.0], 18.0);

    let viewer = Viewer::render(&scene, 800.0, 500.0)?;

    let img = viewer.take_screenshot();

    img.save(Path::new("screenshot_stick.png"))?;

    println!("Press Enter to exit...");
    use std::io::{self, Write};
    let _ = io::stdout().flush();
    let _ = io::stdin().read_line(&mut String::new());

    Ok(())
}
